### ISSUE TITLE
Rate limiting

### ISSUE THREAD

**Issue by JeffProgrammer** [PlatinumTeam/MarbleBot#1]
Discord has a rate limit. We will probably have to queue up chat messages for X number of seconds and then blast them off into the discord server. We don't want to submit a message for every server.

**Comment by hPerks** [comment]
Why not just handle the rate limits when they happen? When a bot is rate limited, the API gives you back a number of milliseconds to wait before sending again; you can just put things on a queue then, instead of doing it all the time. Seems needlessly slow for people who might be doing quick running commentary or something (cough tourney matches).

https://github.com/hammerandchisel/discord-api-docs/blob/master/docs/topics/Rate_Limits.md

**Comment by JeffProgrammer** [comment]
my concern is that the discord system is intelligent enough to be able to determine who keeps hitting the rate limit and apply even higher ones to the point where it is limited badly. Even if they don't do that know, who knows when/if. And tbh i'd rather not find out.

**Comment by hPerks** [comment]
Good point - maybe we should use the rate limit headers they give you normally then to make sure things don't go too fast. For one thing, I don't want to start displaying multiple messages at a time and get this alignment problem, if we don't have to:

![image](https://user-images.githubusercontent.com/19213919/29343109-6123700e-8205-11e7-84d7-82c4fd5cb58e.png)
